libs/telemetry/src/main/java/org/opensearch/telemetry/metrics/MetricsRegistry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.metrics;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+import java.io.Closeable;
+
+/**
+ * MetricsRegistry helps in creating the metric instruments.
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface MetricsRegistry extends Closeable {
```

Comment:
We've introduced a new MetricsRegistry which individual classes may need to inject along with the previously created Tracer interface.

Is it possible to merge the needed interfaces to emit any tracing/metrics, etc. into a single wrapper which can help to avoid a lot of changes if a component decides to have only tracing today, but metrics in future or vice versa.

Reply:
> Is it possible to merge the needed interfaces to emit any tracing/metrics, etc. into a single wrapper which can help to avoid a lot of changes if a component decides to have only tracing today, but metrics in future or vice versa.

Thanks @mgodwan  we have not done metrics injection yet, but I think injecting `Telemetry` or `TelemetryService` instead would make sense (and cover your concerns).